adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java

```diff
@@ -530,6 +532,96 @@ public void setOpenInvoiceData(PaymentRequest paymentRequest, CartData cartData,
         paymentRequest.setInvoiceLines(invoiceLines);
     }
 
+
+    /**
+     * Set the required fields for using the OpenInvoice API
+     */
+    public void setOpenInvoiceData(PaymentsRequest paymentsRequest, CartData cartData, final CustomerModel customerModel) {
+        paymentsRequest.setShopperName(getShopperNameFromAddress(cartData.getDeliveryAddress()));
+
+        // set date of birth
+        if (cartData.getAdyenDob() != null) {
+            paymentsRequest.setDateOfBirth(cartData.getAdyenDob());
+        }
+
+        if (cartData.getAdyenSocialSecurityNumber() != null && ! cartData.getAdyenSocialSecurityNumber().isEmpty()) {
+            paymentsRequest.setSocialSecurityNumber(cartData.getAdyenSocialSecurityNumber());
+        }
+
+        if (cartData.getAdyenDfValue() != null && ! cartData.getAdyenDfValue().isEmpty()) {
+            paymentsRequest.setDeviceFingerprint(cartData.getAdyenDfValue());
+        }
+
+        // set the invoice lines
+        List<LineItem> invoiceLines = new ArrayList<>();
+        String currency = cartData.getTotalPrice().getCurrencyIso();
+
+        for (OrderEntryData entry : cartData.getEntries()) {
+            // Use totalPrice because the basePrice does include tax as well if you have configured this to be calculated in the price
+            BigDecimal pricePerItem = entry.getTotalPrice().getValue().divide(new BigDecimal(entry.getQuantity()));
+
+            String description = "NA";
+            if (entry.getProduct().getName() != null && ! entry.getProduct().getName().isEmpty()) {
+                description = entry.getProduct().getName();
+            }
+
+            // Tax of total price (included quantity)
+            Double tax = entry.getTaxValues().stream().map(TaxValue::getAppliedValue).reduce(0.0, (x, y) -> x + y);
+
+            // Calculate Tax per quantitiy
+            if (tax > 0) {
+                tax = tax / entry.getQuantity().intValue();
```

Comment:
Can the entry.getQuantity().intValue() be zero in any case?

Reply:
shouldn't but prevented in L560